vm/systemSmartContracts/delegationManager.go

```diff
@@ -240,6 +259,50 @@ func (d *delegationManager) createNewDelegationContract(args *vmcommon.ContractC
 	return vmcommon.Ok
 }
 
+func (d *delegationManager) makeNewContractFromValidatorData(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
+	if !d.flagValidatorToDelegation.IsSet() {
+		d.eei.AddReturnMessage("invalid function to call")
+		return vmcommon.UserError
+	}
+	if args.CallValue.Cmp(zero) != 0 {
+		d.eei.AddReturnMessage("callValue must be 0")
+		return vmcommon.UserError
+	}
+	err := d.eei.UseGas(d.gasCost.MetaChainSystemSCsCost.ValidatorToDelegation)
+	if err != nil {
+		d.eei.AddReturnMessage(err.Error())
+		return vmcommon.UserError
+	}
+	if len(args.Arguments) != 2 {
+		d.eei.AddReturnMessage("invalid number of arguments")
+		return vmcommon.UserError
+	}
+
+	arguments := append([][]byte{args.CallerAddr}, args.Arguments...)
+
+	return d.deployNewContract(args, false, "initFromValidatorData", arguments)
```

Comment:
maybe define a const for the `initFromValidatorData` ?

Reply:
done